.github/workflows/cd.yml

```diff
@@ -0,0 +1,56 @@
+name: Deploy the static web app
+on: [push]
+
+jobs:
+    deploy:
+        runs-on: ubuntu-latest
+        env:
+            ARM_CLIENT_ID: ${{ secrets.ARM_CLIENT_ID }}
+            ARM_CLIENT_SECRET: ${{ secrets.ARM_CLIENT_SECRET }}
+            ARM_TENANT_ID: ${{ secrets.ARM_TENANT_ID }}
+            ARM_SUBSCRIPTION_ID: ${{ secrets.ARM_SUBSCRIPTION_ID }}
+            APP_NAME: devops-faqs-static-web-app
+            RESOURCE_GROUP: Delivery_Resources
+            FUNCTION_RESOURCE_GROUP: Delivery_Functions
+            FUNCTION_NAME: static-web-auth-function
+            TF_VAR_GITHUB_CLIENT_ID: ${{ secrets.OAUTH_CLIENT_ID }}
+            TF_VAR_GITHUB_CLIENT_SECRET: ${{ secrets.OAUTH_CLIENT_SECRET }}
+        steps:
+        - uses: actions/checkout@v3
+        
+        - name: Initialise Terraform
+          run: terraform init
+          working-directory: infrastructure
+        - name: Create the infrastructure
+          run: terraform apply -auto-approve
+          working-directory: infrastructure
+
+        - name: Install static web app CLI
+          run: npm install -g @azure/static-web-apps-cli
+        - name: Install credential helper
+          run: sudo apt install gnome-keyring pass
```

Comment:
Assuming this is needed for the `swa deploy` command below?

Reply:
It was a suggested solution, I think I eventually determined not required, but not following up on this repo so will leave here